src/xdp/trn_transit_xdp_hardware_offload.c

```diff
@@ -0,0 +1,580 @@
+					net->switches_ips[swidx]);
+
+		return trn_rewrite_remote_mac(pkt);
+	}
+
+	/* Now forward the packet to the VPC router */
+	struct vpc_key_t vpckey;
+	struct vpc_offload_t *vpc;
+
+	vpckey.tunnel_id = tunnel_id;
+	vpc = bpf_map_lookup_elem(&vpc_map, &vpckey);
+
+	if (!vpc) {
+		return XDP_DROP;
+	}
+
+	/* Only send to the first router. */
+	__u32 routeridx = 0;
+
+	if (routeridx > TRAN_MAX_NROUTER - 1) {
+
+		return XDP_DROP;
+	}
+
+	trn_set_src_dst_ip_csum(pkt, pkt->ip->daddr,
+				vpc->routers_ips[routeridx]);
+
+	return trn_rewrite_remote_mac(pkt);
+}
+
+
+static __inline int trn_switch_handle_pkt(struct transit_packet *pkt,
+					  __u32 inner_src_ip,
+					  __u32 inner_dst_ip, __u32 orig_src_ip)
+{
+	__be64 tunnel_id = trn_vni_to_tunnel_id(pkt->geneve->vni);
+	struct endpoint_offload_t *ep;
+	struct endpoint_key_t epkey;
+
+	__builtin_memcpy(&epkey.tunip[0], &tunnel_id, sizeof(tunnel_id));
+	epkey.tunip[2] = inner_dst_ip;
+
+	/* Get the remote_ip based on the value of the inner dest IP and VNI*/
+	ep = bpf_map_lookup_elem(&endpoints_map, &epkey);
+
+	if (!ep) {
+		if (pkt->scaled_ep_opt->type == TRN_GNV_SCALED_EP_OPT_TYPE &&
+		    pkt->scaled_ep_opt->scaled_ep_data.msg_type ==
+			    TRN_SCALED_EP_MODIFY)
+			return XDP_PASS;
+		
+		return trn_router_handle_pkt(pkt, inner_src_ip, inner_dst_ip);
+	}
+
+	/* The packet may be sent first to a gw mac address */
+	trn_set_dst_mac(pkt->inner_eth, ep->mac);
+
+	// TODO: Currently all endpoints are attached to one host, for some
+	// ep types, they will have multiple attachments (e.g. LB endpoint).
+	if (ep->hosted_iface != -1) {
+		return XDP_PASS;
+	}
+
+	if (ep->eptype == TRAN_SCALED_EP) {
+		return XDP_PASS;
+	}
+
+	if (ep->nremote_ips == 0) {
+
+		return XDP_DROP;
+	}
+
+	trn_set_src_dst_ip_csum(pkt, pkt->ip->daddr, ep->remote_ips[0]);
+
+	return trn_rewrite_remote_mac(pkt);
+}
+
+static __inline int trn_process_inner_ip(struct transit_packet *pkt)
+{
+	pkt->inner_ip = (void *)pkt->inner_eth + pkt->inner_eth_off;
+	__u32 ipproto;
+
+	if (pkt->inner_ip + 1 > pkt->data_end) {
+
+		return XDP_ABORTED;
+	}
+
+	/* For whatever compiler reason, we need to perform reverse flow modification
+	 in this function instead of trn_switch_handle_pkt so we keep the orig_src_ip */
+	__u32 orig_src_ip = pkt->inner_ip->saddr;
+
+	pkt->inner_ipv4_tuple.saddr = pkt->inner_ip->saddr;
+	pkt->inner_ipv4_tuple.daddr = pkt->inner_ip->daddr;
+	pkt->inner_ipv4_tuple.protocol = pkt->inner_ip->protocol;
+	pkt->inner_ipv4_tuple.sport = 0;
+	pkt->inner_ipv4_tuple.dport = 0;
+
+	if (pkt->inner_ipv4_tuple.protocol == IPPROTO_TCP) {
+		pkt->inner_tcp = (void *)pkt->inner_ip + sizeof(*pkt->inner_ip);
+
+		if (pkt->inner_tcp + 1 > pkt->data_end) {
+
+			return XDP_ABORTED;
+		}
+
+		pkt->inner_ipv4_tuple.sport = pkt->inner_tcp->source;
+		pkt->inner_ipv4_tuple.dport = pkt->inner_tcp->dest;
+	}
+
+	if (pkt->inner_ipv4_tuple.protocol == IPPROTO_UDP) {
+		pkt->inner_udp = (void *)pkt->inner_ip + sizeof(*pkt->inner_ip);
+
```

Comment:
please remove spurious blank lines and follow convention iwth rest of xdp code, be consistent.